src/gui/dialogs/addon/manager.cpp

```diff
@@ -391,6 +391,37 @@ void addon_manager::pre_show(window& window)
 	connect_signal_notify_modified(type_filter,
 		std::bind(&addon_manager::apply_filters, this));
 
+	// Language filter
+	// Prepare shown languages, source all available languages from the addons themselves
+	std::set<std::string> languages_available;
+	for(const auto& a : addons_) {
+		for (const auto& b : a.second.locales) {
+			languages_available.insert(b);
+		}
+	}
+
+	// Erase pt value, since otherwise Portugues do Brasil will be shown twice in the language list
+	languages_available.erase(languages_available.find("pt"));
```

Comment:
I'd prefer this to be some logic instead of a hardcoded list, because there will eventually be two `es` languages too.

Reply:
As discussed on Discord: it would be better to have the language display name inserted into the set rather than hard coding the removal of "pt".

So change
`languages_available.insert(b);`
to
`languages_available.insert(langcode_to_string(b));`
and then erase with:
`languages_available.erase(languages_available.find(""));`

Reply:
I definitely don't think hard-coding a list is appropriate here, as there are any number of language codes that have variants. For example, fr_FR vs fr_CA, to name a common one. (Wesnoth may not have both _now_, but there's no reason to assume it will never have both.)

That said, you do have a point that Simplified vs Traditional Chinese are different enough that people might want to distinguish. This is kind of a difficult problem. I definitely think that filtering by language alone is a more sensible default, but it's also not terrible to support filtering on language plus locale as an option. That is quite a huge complication to this system, though – I think to really handle it well you'd need a completely new menu type (where the menu contents is a tree view similar to what you see sometimes in an app installer, with tri-state checkboxes on the non-lead nodes and regular dual-state checkboxes on the leaves).

So doing both is definitely out of scope for this pull request. This can be thought of as a first iteration of the language filtering system, so I'd say it should either:

1. Just filter on full locale – don't automatically merge pt_BR and pt_PT or anything like that.
2. Filter only on language, but add a hard-coded exception for (at least) Chinese so they can be filtered separately.

Just having the ability to filter by language is already great, and the system can be further refined later. I actually think being able to set an add-on's default locale is more important than limiting the filter to language alone without considering the full locale.

Reply:
Guys, I found something I should have done a long time ago that fixes all these problems: getting ALL languages regardless of translation percentage. Normally only languages are shown that only exceed a certain translation threshold, but for the language filter case that should not matter. Therefore I show now all possible localisations (see pictures)
![LangFilterNow](https://user-images.githubusercontent.com/26515148/199960071-851a32ce-65f5-4eee-86f7-49684823d72a.jpeg)

@Pentarctagon that turned out to be inviting some faulty behaviour in the GUI (see picture), so I kept the old implementation, which somehow eliminates that behaviour
![LangFilterBroken](https://user-images.githubusercontent.com/26515148/199960067-f586dd11-3fdc-44ae-b4c1-4c2239522735.jpeg)

Reply:
I'm not sure what issue the image is meant to show?

Reply:
The language button dropdown menu fills the entire screen and is empty

Reply:
Huh, well that's unexpected. Anyway, if the current code in the PR works, that's good then.